.devcontainer/README.md

```diff
@@ -0,0 +1,15 @@
+# Development Container
+
+This directory contains all files which pertain to this project's development container. The development container has been tested with [VSCode](https://code.visualstudio.com).
+
+To use the development container environment simply open the project's top level directory with VSCode. Consider also [these](https://code.visualstudio.com/docs/remote/containers#_installation) instructions. This environment should work for MacOS, Windows, and Linux host systems.
+
+## Build Docker Image
+
+Building the docker image is usually unnecessary for most developers. Instead, VSCode will automatically retrieve an existing docker image from Docker Hub based on the `image` field in `.devcontainer/devcontainer.json`.
+
+The build process can take ~40mins or more. The development container's base docker image can be built by executing this command in the project's root directory:
```

Comment:
Wow, do we know what takes so long? Can this be sped up using cachix? (although that might be beyond the point of building yourself)

Reply:
The first execution of the command `nix develop` has the longest runtime:
https://github.com/fedimint/fedimint/pull/353/commits/83a9732fd59dac1756f34c0e8b2135be4990d7d3#diff-13bd9d7a30bf46656bc81f1ad5b908a627f9247be3f7d76df862b0578b534fc6R29

Subsequent calls (such as from `.bashrc`) take less than a second.

40mins was an estimate. But it's a non-trivial amount of time.

Reply:
This seems very long. Most of the Nix stuff should be downloaded from NixOS binary cache, and then some custom parts that can't - take maybe 20 minutes? 

Is the build time CPU or network bound?

Reply:
It's an estimate from glancing at my screen every now and then. I though 40mins might give a potential docker image builder the right expectation.

I didn't think it would matter. But I will measure it so that we have an accurate figure to work with.

Reply:
We could eventually have CI build this container. Then user could just pull it from dockerhub.